File: analise-e-projetos-de-algoritmo/Lista 03/organizador de vagoes.cpp
#include <iostream>

using namespace std;

int main()
{

    int testes;

    cin >> testes;

    for(int i = 0; i< testes; i++)
    {

        int tamanho;

        cin >> tamanho;

        int vagoes[tamanho];

        int swaps = 0;

        for(int j = 0; j< tamanho; j++)
        {

            cin >> vagoes[j];

        }

        int aux;

        for(int i=0; i<tamanho-1; i++){
            for(int j = i + 1; j<tamanho; j++){
                if(vagoes[i] > vagoes[j]){
                    aux = vagoes[j];
                    vagoes[j] = vagoes[i];
                    vagoes[i] = aux;
                    swaps++;
                }
            }
        }

        cout << "Optimal train swapping takes " << swaps << " swaps." << endl;
    }

}

File: analise-e-projetos-de-algoritmo/Lista 03/organizador de vagoes.py

testes = input()
testes = int(testes)

for i in range(testes):
    tamanho = input()
    tamanho = int(tamanho)

    vagoes = [int(x) for x in input().split()]

    #print(vagoes)

    swaps = 0

    for j in range(tamanho-1):
        for k in range(j+1,tamanho):
            if vagoes[j] > vagoes[k]:
                auxiliar = vagoes[j]
                vagoes[j] = vagoes[k]
                vagoes[k] = auxiliar
                swaps +=1
            
            #print(vagoes, swaps)
    
    print("Optimal train swapping takes",swaps,"swaps.")
    